gym/envs/mujoco/mujoco_env.py

```diff
@@ -391,9 +396,7 @@ def close(self):
             self.viewer.close()
         super().close()
 
-    def _get_viewer(
-        self, mode
-    ) -> Union["gym.envs.mujoco.Viewer", "gym.envs.mujoco.RenderContextOffscreen"]:
+    def _get_viewer(self, mode) -> Union[Viewer, RenderContextOffscreen]:
```

Comment:
If you uninstall mujoco_py, then an error is raised due to this line I believe.
To replicate
`pip uninstall mujoco_py`
`import gym`
`gym.make("Ant-v4")`
`pip install "gym[mujoco_py]"`

Reply:
Yes, I've confirmed that it raises an error. Thanks for catching this. 
A solution would be to use;
```python
if TYPE_CHECKING:
    from gym.envs.mujoco import RenderContextOffscreen, Viewer
```
and 
```python
def _get_viewer(self, mode) -> Union["Viewer", "RenderContextOffscreen"]:
```
I'll now update this.

Reply:
There is a `ModuleNotFoundError` error when mujoco is uninstalled but it's not related to type checking.

Reply:
That is because v4 environment use mujoco from while v2 and v3 environments use mujoco_py. Could you test with `Ant-v2`

Reply:
I get the same mujoco `ModuleNotFoundError`, coming from `gym\envs\mujoco\mujoco_rendering.py` when it is imported in `gym/envs/mujoco/__init__.py`

I can either remove the `mujoco_rendering` import in `__init__.py` or use a try-except in `mujoco_rendering.py` where mujoco is imported.

Reply:
I think for now we don't worry about it and revert to the old implementation with type: ignores

Reply:
Okay